app/client/components/GridView.js

```diff
@@ -902,6 +904,34 @@ GridView.prototype.insertColumn = async function(colId = null, options = {}) {
   return newColInfo;
 };
 
+GridView.prototype.makeHeadersFromRow = async function(selection) {
+  if (this._getCellContextMenuOptions().disableMakeHeadersFromRow){
+    return;
+  }
+  const record = this.tableModel.tableData.getRecord(selection.rowIds[0]);
+  const actions = this.viewSection.viewFields().peek().reduce((acc, field) => {
+    const col = field.column();
+    const colId = col.colId.peek();
+    let formatter = field.formatter();
+    let newColLabel = record[colId];
+    // Manage column that are references
+    if (col.refTable()) {
+      const refTableDisplayCol = this.gristDoc.docModel.columns.getRowModel(col.displayCol());
+      newColLabel =  record[refTableDisplayCol.colId()];
+      formatter = formatter.visibleColFormatter;
```

Comment:
I don't understand your remark because the code do this. See in the screen shot. I put a breakpoint inside reference if statement and a do some logs
![Screenshot from 2024-03-07 09-24-53](https://github.com/gristlabs/grist-core/assets/19594405/22800618-8d33-4638-98ac-52eeff86ae08)
![Screenshot from 2024-03-07 09-25-27](https://github.com/gristlabs/grist-core/assets/19594405/61fad995-7949-4196-b775-64e271811106)


Or you're talking about a reference list of a reference list ?

Reply:
Sorry, you're right. It's working well for me on my end too. Just remembered why this also works.